keras_nlp/layers/preprocessing/mlm_mask_generator.py

```diff
@@ -0,0 +1,216 @@
+        numpy=array([[1, 0, 0, 0, 0]])>}
+
+    Ragged Input:
+    >>> masker = keras_nlp.preprocessing.MLMMaskGenerator( \
+    ...     10, mask_selection_rate=0.5, mask_selection_length=5)
+    >>> masker(tf.ragged.constant([[1, 2], [1, 2, 3, 4]]))
+    {'tokens': <tf.RaggedTensor [[1, 4], [0, 2, 3, 0]]>,
+    'mask_positions': <tf.Tensor: shape=(2, 5), dtype=int64, numpy=
+        array([[1, 0, 0, 0, 0],
+               [0, 3, 0, 0, 0]])>,
+    'mask_ids': <tf.Tensor: shape=(2, 5), dtype=int32, numpy=
+        array([[2, 0, 0, 0, 0],
+               [1, 4, 0, 0, 0]], dtype=int32)>,
+    'mask_weights': <tf.Tensor: shape=(2, 5), dtype=int64, numpy=
+        array([[1, 0, 0, 0, 0],
+               [1, 1, 0, 0, 0]])>}
+    """
+
+    def __init__(
+        self,
+        vocabulary_size,
+        mask_selection_rate,
+        mask_selection_length=None,
+        unselectable_token_ids=None,
+        mask_token_id=0,
+        mask_token_rate=0.8,
+        random_token_rate=0.1,
+        padding_token_id=None,
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
+        self.vocabulary_size = vocabulary_size
+        self.unselectable_token_ids = unselectable_token_ids
+        self.mask_selection_rate = mask_selection_rate
+        self.mask_selection_length = mask_selection_length
+        self.mask_token_rate = mask_token_rate
+        self.random_token_rate = random_token_rate
+        self.padding_token_id = padding_token_id
+
+        if mask_token_id >= vocabulary_size:
+            raise ValueError(
+                f"Mask token id should be in range [0, vocabulary_size - 1], "
+                f"but received mask_token_id={mask_token_id}."
+            )
+        self.mask_token_id = mask_token_id
+
+        max_selections = self.mask_selection_length
+        if max_selections is None:
+            # Set a large number to remove the `max_selections_per_batch` cap.
+            max_selections = 2**31 - 1
+        self._random_selector = tf_text.RandomItemSelector(
+            max_selections_per_batch=max_selections,
+            selection_rate=self.mask_selection_rate,
+            unselectable_ids=self.unselectable_token_ids,
+        )
+        self._mask_values_chooser = tf_text.MaskValuesChooser(
+            self.vocabulary_size,
+            self.mask_token_id,
+            mask_token_rate=self.mask_token_rate,
+            random_token_rate=self.random_token_rate,
+        )
+
+    def call(self, inputs):
+        input_is_ragged = isinstance(inputs, tf.RaggedTensor)
+        input_is_1d = tf.rank(inputs) == 1
+        if input_is_1d:
+            # If inputs is of rank 1, we manually add the batch axis.
+            inputs = inputs[tf.newaxis, :]
+        if not input_is_ragged:
+            # Convert to RaggedTensor to avoid masking out padded token.
+            inputs = tf.RaggedTensor.from_tensor(
+                inputs,
+                padding=self.padding_token_id,
```

Comment:
If you take my suggestion above (just removing padding_token_id), we can just remove the padding term here. That's ok--we will still mark the padding terms unselectable by default via `unselectable_token_ids`.